api/types/registry/registry.go

```diff
@@ -102,3 +105,18 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// ManifestInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+// TODO(nishanttotla): Decide which other fields to add here
+type ManifestInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
+	// Platforms contains the list of platforms supported by the image,
+	// possibly obtained by looking at the fat manifest
+	Platforms []manifestlist.PlatformSpec
+	// Manifest contains the original manifest JSON
+	Manifest *json.RawMessage
```

Comment:
I guess since we're potentially returning different kinds of manifests in this field (schema1, schema2, manifest list...), we should have another field containing the media type. Depending which kind of manifest is being returned, this field would either be set to `schema1.MediaTypeSignedManifest`, `schema2.MediaTypeManifest`, or `manifestlist.MediaTypeManifestList`.